common/remote/sockfile.cpp

```diff
@@ -4020,20 +4171,216 @@ IRemoteActivity *createRemoteDiskRead(IPropertyTree &actNode)
     if (!outMeta)
         outMeta.set(inMeta);
     Owned<IHThorDiskReadArg> helper = createDiskReadArg(fileName, LINK(inMeta), LINK(outMeta), LINK(outMeta), chooseN, skipN, (unsigned __int64)-1);
-    Owned<CRemoteDiskReadActivity> ret = new CRemoteDiskReadActivity(*helper, compressed, inputGrouped, outputGrouped);
-    Owned<IPropertyTreeIterator> filterIter = actNode.getElements("keyfilter");
+    assertex(!helper->needTransform()); // Code assumes that we do not need to call the transform
+    Owned<CRemoteDiskReadActivity> ret = new CRemoteDiskReadActivity(helper, compressed, inputGrouped, outputGrouped);
+    Owned<IPropertyTreeIterator> filterIter = actNode.getElements("keyFilter");
     ForEach(*filterIter)
         ret->addFilter(filterIter->query().queryProp(nullptr));
     return ret.getClear();
 }
 
+class CRemoteIndexReadActivity : public CRemoteDiskBaseActivity
+{
+    typedef CRemoteDiskBaseActivity PARENT;
+
+protected:
+    Linked<IHThorIndexReadArg> helper;
+
+    bool isTlk = false;
+    bool allowPreload = false;
+    unsigned __int64 seeks = 0;
+    unsigned __int64 scans = 0;
+    Owned<IKeyIndex> keyIndex;
+    Owned<IKeyManager> keyManager;
+
+    void checkOpen()
+    {
+        if (opened)
+            return;
+        if (canMatchAny)
+        {
+            const char *fileName = helper->getFileName();
+
+            keyIndex.setown(createKeyIndex(fileName, 0, isTlk, allowPreload)); // JCSMORE crc
+            keyManager.setown(createLocalKeyManager(*record, keyIndex, nullptr, helper->hasNewSegmentMonitors()));
+            helper->createSegmentMonitors(keyManager);
+            keyManager->finishSegmentMonitors();
+            keyManager->reset();
+
+            chooseN = helper->getChooseNLimit();
+            opened = true;
+        }
+        else
+            eofSeen = true;
+    }
+    void close()
+    {
+        keyManager.clear();
+        keyIndex.clear();
+        opened = false;
+        eofSeen = true;
+    }
+public:
+    CRemoteIndexReadActivity(IHThorIndexReadArg *_helper, bool _isTlk, bool _allowPreload) : PARENT(_helper, _helper->queryDiskRecordSize(), _helper->queryProjectedDiskRecordSize()),
+        isTlk(_isTlk), allowPreload(_allowPreload), helper(_helper)
+    {
+    }
+    IMPLEMENT_IINTERFACE_USING(CSimpleInterfaceOf<IRemoteActivity>)
```

Comment:
This line should really be in the baseclass CRemoteDiskBaseActivity

Reply:
looks like it hasn't been deleted.

Reply:
will delete